From c509dfc721603b71cd5e6ca61014f1e68add8506 Mon Sep 17 00:00:00 2001
From: Vladimir Oltean <olteanv@gmail.com>
Date: Tue, 25 Sep 2018 14:40:30 +0300
Subject: [PATCH 18/18] legacy: allow init to read /proc/device-tree

* Solves denials:

  avc: denied { read } for pid=1 comm="init" name="compatible" dev="proc"
       scontext=u:r:init:s0 tcontext=u:object_r:proc:s0 tclass=file permissive=1
  avc: denied { open } for pid=1 comm="init" path="/proc/device-tree/firmware/android/compatible"
       dev="proc" scontext=u:r:init:s0 tcontext=u:object_r:proc:s0 tclass=file permissive=1
  avc: denied { getattr } for  pid=1 comm="init" path="/proc/device-tree/firmware/android/compatible"
       dev="proc" scontext=u:r:init:s0 tcontext=u:object_r:proc:s0 tclass=file permissive=1

Change-Id: Ib5543f3f4d0e6a723a0f45621d607f9b9a25d08f
Signed-off-by: Vladimir Oltean <olteanv@gmail.com>
---
 legacy-common/genfs_contexts | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/legacy-common/genfs_contexts b/legacy-common/genfs_contexts
index f94e9eb..41bbf9e 100644
--- a/legacy-common/genfs_contexts
+++ b/legacy-common/genfs_contexts
@@ -1 +1,2 @@
-genfscon debugfs /rmt_storage u:object_r:debugfs_rmt:s0
+genfscon debugfs /rmt_storage                    u:object_r:debugfs_rmt:s0
+genfscon proc    /device-tree/firmware/android   u:object_r:sysfs_dt_firmware_android:s0
-- 
2.17.1

